.github/workflows/main.yml

```diff
@@ -0,0 +1,39 @@
+# This is a basic workflow to help you get started with Actions
+
+name: CI
+
+# Controls when the workflow will run
+on:
+  push:
+    branches: [ "main", "master" ]
+  schedule:
+    - cron:  '*/15 * * * *'
+  workflow_dispatch:
+
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/checkout@v3
+        with:
+          ref: "hp-pages"
+          path: "pages"
+      - uses: actions/checkout@v3
+        with:
+          repository: "SerenityOS/serenity"
+          path: "serenity"
+          fetch-depth: "0"
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.8"
+          cache: 'pip' # caching pip dependencies
+      - run: python3 -m pip install -r requirements.txt
```

Comment:
Why set up pip and add a requirements file? That file is empty, so this call does absolutely nothing.

Reply:
As you said, It's useless here, just provisions for a future time were some external modules could be needed.